.github/workflows/pr-lint.yml

```diff
@@ -0,0 +1,16 @@
+name: PR Quality Check
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+jobs:
+  check-pr-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check ticket name conforms to requirements
+        run: 'echo ${{ github.event.pull_request.title }} | grep -i -E -q "^((apm|niad)-[0-9]+): ([a-zA-Z]+)|(dependabot\/)"'
```

Comment:
We can probably be confident we'll only be working in the NIAD project, so APM is not needed. Check with @bartek-sarul ?

Also the `-i` flag is "ignore case".
```
> man grep
/-i
```
```
    -i, --ignore-case
             Perform case insensitive matching.  By default, grep is case sensitive.
```
so the character set doesn't need to include both cases.
(This latter point applies to line 16 too.)